### ISSUE TITLE
Select all is not updating when all resources are selected

### ISSUE THREAD

**Issue by mlabouardy** [tailwarden/komiser#359]
When a user selects all the resources, the select all checkbox should update to the "all selected" state.

![image](https://user-images.githubusercontent.com/10320205/218456154-cbdb3ccc-fb9c-4a48-8fc0-5a407a67d3df.png)

It also happens when only a few resources are selected, it should update to the "semi selected" state.

**Comment by ShivamTyagi12345** [comment]
Let me work on this one  ( 👍

**Comment by mlabouardy** [comment]
ofc, just assigned to you @ShivamTyagi12345

**Comment by ShivamTyagi12345** [comment]
hello @mlabouardy , can u please look why doesnt the frontend fire up when i am running the right commands as mentioned in the contributing guideline

![image](https://user-images.githubusercontent.com/60812924/219369032-12e3f688-2583-4b7e-b967-1f13416f1712.png)
![image](https://user-images.githubusercontent.com/60812924/219369078-c69cad7f-48cd-4b2b-b547-69db7ba41433.png)

**Comment by mlabouardy** [comment]
hey @ShivamTyagi12345 you need to run the backend server and point the frontend (aka dashboard) to the backend server with the following:
```
# From the Komiser root folder start the Komiser server, run:
go run *.go start --config /path/to/config.toml

# In a different terminal tab in the dashboard folder, run:
NEXT_PUBLIC_API_URL=http://localhost:3000 npm run dev
```

**Comment by ShivamTyagi12345** [comment]
> go run *.go start --config /path/to/config.toml

There are two things i wanted help with:
- After running, `go run *.go start --config /path/to/config.toml ` in the komiser root directory, i get the following error 


`:~/Documents/Tailwarden/komiser$ go run *.go start --config /path/to/config.toml
INIT!!!
INFO[2023-02-16T18:31:03+05:30] Debug logging is enabled                     
Error: no such file /path/to/config.toml
Usage:
  komiser start [flags]

Flags:
      --config string           Path to configuration file. (default "config.toml")
  -h, --help                    help for start
  -a, --listen-address string   Listen address to start server on. (default "0.0.0.0")
      --no-tracking             Disable user analytics.
      --port int                Port to start server on, default:"3000". (default 3000)
      --regions stringArray     Restrict Komiser inspection to list of regions.
      --verbose                 Show verbose debug information. (default true)

exit status 1`


TL/DR .. i wanted to ask if i had to run the command `NEXT_PUBLIC_API_URL=http://localhost:3000 npm run dev` after i have ran the command `npm run dev` in a seperate terminal?

**Comment by jakepage91** [comment]
Hey there @ShivamTyagi12345 make sure you change the path to the location your config.toml is on your local machine `--config /path/to/config.toml` 

The error you shared above indicates that the path to the config.toml file is incorrect. 

**And for your second question:** 
Yes you are going to run `NEXT_PUBLIC_API_URL=http://localhost:3000 npm run dev` in a separate terminal window or tab and then open `localhost:3000`. 

Let me know if that works for you 👍

**Comment by ShivamTyagi12345** [comment]
thanks fgr that response Jake, I do know that we can find any file in ubuntu with this command `find . -name config.toml` , however i wasnt able to find the location of the config.toml file as it returned back no o/p
![image](https://user-images.githubusercontent.com/60812924/219386920-04781d92-2977-4427-9d40-3d7add5ce6a6.png)

I spent a fair amount of time manually searching it as well. Can u please let me know how can i locate the same?

**Comment by jakepage91** [comment]
Hey @ShivamTyagi12345, you will have to create it yourself since it's the mechanism with which we authenticate to our cloud accounts. 
Do you have any cloud accounts at the moment? For example, an AWS free tier account or a Civo account for example. If not, you will have to open one (this shouldn't cost you any money, even though you will have to associate a credit card.) And then create the config file. I have my personal config.toml file inside a `komiser-local` directory I created and I specify the path to the config file every time I run Komiser from a different directory like so: `komiser start --config /Users/jakpage/komiser-local/config.toml`

Here is an example of a config.toml file for a civo account: 
```
[[civo]]
name="demo-account"
token="yourApiTokenHere"

[sqlite]
file="komiser.db
```
For more info check out this ⬇️ 
https://docs.komiser.io/docs/cloud-providers/civo?utm_source=social&utm_medium=github

**Comment by ShivamTyagi12345** [comment]
> Hey @ShivamTyagi12345, you will have to create it yourself since it's the mechanism with which we authenticate to our cloud accounts. Do you have any cloud accounts at the moment? For example, an AWS free tier account or a Civo account for example. If not, you will have to open one (this shouldn't cost you any money, even though you will have to associate a credit card.) And then create the config file. I have my personal config.toml file inside a `komiser-local` directory I created and I specify the path to the config file every time I run Komiser from a different directory like so: `komiser start --config /Users/jakpage/komiser-local/config.toml`
> 
> Here is an example of a config.toml file for a civo account:
> 
> ```
> [[civo]]
> name="demo-account"
> token="yourApiTokenHere"
> 
> [sqlite]
> file="komiser.db
> ```
> 
> For more info check out this arrow_down https://docs.komiser.io/docs/cloud-providers/civo?utm_source=social&utm_medium=github

Yes I just got my AWS account set up, and now i will figure out how to create the config,toml file. I will be referring to :arrow_down: 
https://docs.komiser.io/docs/cloud-providers/aws
If there is any resource that can make my job easy please let me know

**Comment by mlabouardy** [comment]
Hey @ShivamTyagi12345 were you able to tackle the issue? :) let me know if you need a hand on this

**Comment by ShivamTyagi12345** [comment]
> Hey @ShivamTyagi12345 were you able to tackle the issue? :) let me know if you need a hand on this

Thanks for the follow up , but i am still figuring out how to find the content of config.toml using my AWS account. I read the documentation but maybe i need to look into couple more of them , Can u please provide me with some help

**Comment by jakepage91** [comment]
Hey there @ShivamTyagi12345 the easiest way would be to: 

- Create an AWS free tier account
- Install [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html)
- Configure your [credentials](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html)  
- Locate your credentials in your local `~/.aws/credentials` file.
- Create a config.toml file and include a path to the credentials file.

The credentials file should look like this: 
```
[default]
aws_access_key_id = <access key id>
aws_secret_access_key = <secret access key>
region = <AWS region>
```
The Config.toml file like this 
```
[[aws]]
name="sandbox"
source="CREDENTIALS_FILE"
path=./path/to/credentials/file  <<<<<<<<<< CHANGE THIS LINE
profile="default"
```

If you need any help with this ping me on discord and we can jump into one of the voice rooms and I can share me screen and walk you through the process.

**Comment by ShivamTyagi12345** [comment]
> Hey there @ShivamTyagi12345 the easiest way would be to:
> 
>     * Create an AWS free tier account
> 
>     * Install [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html)
> 
>     * Configure your [credentials](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html)
> 
>     * Locate your credentials in your local `~/.aws/credentials` file.
> 
>     * Create a config.toml file and include a path to the credentials file.
> 
> 
> The credentials file should look like this:
> 
> ```
> [default]
> aws_access_key_id = <access key id>
> aws_secret_access_key = <secret access key>
> region = <AWS region>
> ```
> 
> The Config.toml file like this
> 
> ```
> [[aws]]
> name="sandbox"
> source="CREDENTIALS_FILE"
> path=./path/to/credentials/file  <<<<<<<<<< CHANGE THIS LINE
> profile="default"
> ```
> 
> If you need any help with this ping me on discord and we can jump into one of the voice rooms and I can share me screen and walk you through the process.

Just a follow up question for step3 , i cannot see a .confifg or credentials file in my aws folder
![image](https://user-images.githubusercontent.com/60812924/220371943-d8a489ad-7478-4e4f-a469-92b378fcd6a6.png)

**Comment by jakepage91** [comment]
You generate the credentials file by passing in `aws configure` in your local terminal 

Example:
```
$ aws configure
AWS Access Key ID [None]: AKIAIOSFODNN7EXAMP <<< change this
AWS Secret Access Key [None]: wJalrXUtnFEMI/K7ENG/bPxRfiCYEXAMPLEKEY <<< change this
Default region name [None]: us-west-2 <<< change this
Default output format [None]: json
```

Check out the instructions here: https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html

**Comment by mlabouardy** [comment]
> > Hey there @ShivamTyagi12345 the easiest way would be to:
> > ```
> > * Create an AWS free tier account
> > 
> > * Install [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html)
> > 
> > * Configure your [credentials](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html)
> > 
> > * Locate your credentials in your local `~/.aws/credentials` file.
> > 
> > * Create a config.toml file and include a path to the credentials file.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The credentials file should look like this:
> > ```
> > [default]
> > aws_access_key_id = <access key id>
> > aws_secret_access_key = <secret access key>
> > region = <AWS region>
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The Config.toml file like this
> > ```
> > [[aws]]
> > name="sandbox"
> > source="CREDENTIALS_FILE"
> > path=./path/to/credentials/file  <<<<<<<<<< CHANGE THIS LINE
> > profile="default"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > If you need any help with this ping me on discord and we can jump into one of the voice rooms and I can share me screen and walk you through the process.
> 
> Just a follow up question for step3 , i cannot see a .confifg or credentials file in my aws folder ![image](https://user-images.githubusercontent.com/60812924/220371943-d8a489ad-7478-4e4f-a469-92b378fcd6a6.png)

Any update on this issue? :)

**Comment by ShivamTyagi12345** [comment]
Hey @mlabouardy I have been involved with my job lately, so this skipped my attention. Do u mind if i resume on this now

**Comment by mlabouardy** [comment]
> Hey @mlabouardy I have been involved with my job lately, so this skipped my attention. Do u mind if i resume on this now

For sure, let me know if you need a hand on it :)

**Comment by ShubhamPalriwala** [comment]
Hey @ShivamTyagi12345, let me know if you need any help with this? would be great if we can get this ready for the next release 🚀

**Comment by ShivamTyagi12345** [comment]
if possible can we get on a call and finish this @ShubhamPalriwala

**Comment by ShubhamPalriwala** [comment]
Hey @ShivamTyagi12345, that'd be great. Since you are already a part of the Discord server, lets sync up in either of:
- Contributor Office Hours thats happening at 5:30 PM IST tomorrow (technically today)
- Or Friday afternoon IST's Komiser Weekly

If none work for you, lets take this async in the #contributors channel on the server 💯

**Comment by jakepage91** [comment]
Hey there @ShivamTyagi12345 I hope you are doing well! How is this issue going? Feel free to join the Contributor office hours today if you need any help! 

Link to contributor office hours here ==> https://discord.gg/2mHSz8DW?event=1128362990027153439
